clients/src/main/resources/common/message/ListGroupsRequest.json

```diff
@@ -20,8 +20,14 @@
   // Version 1 and 2 are the same as version 0.
   //
   // Version 3 is the first flexible version.
-  "validVersions": "0-3",
+  //
+  // Version 4 adds the StatesFilter field (KIP-518).
+  "validVersions": "0-4",
   "flexibleVersions": "3+",
   "fields": [
+    { "name": "StatesFilter", "type": "[]string", "versions": "4+", "nullableVersions": "4+", "default": "null",
```

Comment:
The behavior of null is identical to an empty list in terms of semantics. However, if an empty list is provided, the request cannot be serialized with older versions because the field is marked as ignorable. Would it be simpler if we made this field non-nullable and used the default of an empty array so that we don't have this inconsistency?